### PR TITLE
PAPP-28938: Converting app to splunk supported app

### PR DESCRIPTION
**Pull request by mmomin-crest**
FYI - The cloud job of the sanity pipeline is failing due to AB issues. AB is taking the older version of the app. Hence, Pytest is failing. We ran the latest app version on the same cloud instance with different AB and it has passed successfully. Manually ran job link - https://us-west-2.console.aws.amazon.com/codesuite/codebuild/875003031410/projects/sanity-tests/build/sanity-tests%3A87412622-bbb3-4899-be28-5b4f26c88011?region=us-west-2

Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [ ] Testing of all the changes has been performed (for bug fixes / features)
- [ ] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [ ] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [ ] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [ ] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [x] Other (please describe): API changes

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers), 
  please document them in the `readme.html`.
- [ ] If this is a new connector or you are adding new actions
    - [ ] Please document in the `readme.html` all methods (eg, OAuth) used to authenticate 
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `readme.html`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through 
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

  

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

File: taniumthreatresponse.json [hunk 1]
```diff
@@ -2822,6 +3441,7 @@
         {
             "action": "start quick scan",
             "description": "Scan a computer group for hashes in intel document",
+            "verbose": "This action requires a 'threat response on demand scans write' permission. If this action is failing with 'GroupNotFound' exception then please try with the ID of computer group for 'computer group name' parameter. You can find Computer Group ID from Tanium UI->Administration->Computer Groups->Click on Computer Group name now you can see the ID on URL. If a numerical value is given to the computer group parameter, then priority will be given to 'ID' and ID is not present in tanium, then consider ID as the name of the computer group. If the computer group is deleted but visible on Tanium through the group ID. In this case, this action will be passed using the deleted group ID and scan will start on Tanium without computer group name.",
```
Comment (alexa-phantom) [inline taniumthreatresponse.json:3444]: Same here

File: taniumthreatresponse_connector.py [hunk 2]
```diff
@@ -464,25 +475,35 @@ def _handle_list_connections(self, param):
...
-            for status in [x['status'] for x in response if 'status' in x]:
-                summary['{}_connections'.format(
-                    CONNECTION_STATUS[status])] = len([x for x in response if 'status' in x and x['status'] == status])
-            for r in response:
-                action_result.add_data(r)
-                summary['total_connections'] = len(response)
+            active_connections = 0
+            inactive_connections = 0
+            for resp in response:
+                if 'status' in resp and resp['status'] == 'disconnected':
```
Comment (alexa-phantom) [inline taniumthreatresponse_connector.py:487]: Could also do `resp.get('status')`?

**Review by alexa-phantom — CHANGES_REQUESTED** [review]

File: taniumthreatresponse.json [hunk 3]
```diff
@@ -33,26 +34,27 @@
...
+            "description": "API Token",
+            "data_type": "password",
             "order": 4
         }
     },
     "actions": [
         {
             "action": "test connectivity",
             "description": "Validate the asset configuration for connectivity using supplied configuration",
+            "verbose": "This action requires a 'threat response use api' permission.",
```
[the remark this answers is not in the document]
Comment (mmomin-crest) [reply taniumthreatresponse.json]: Updated verbose as this action require threat response use api permission.

File: taniumthreatresponse.json (hunk 1 above)
Comment (mmomin-crest) [reply taniumthreatresponse.json]: Updated verbose as this action require 'threat response on demand scans write' permission.

File: taniumthreatresponse_connector.py (hunk 2 above)
Comment (mmomin-crest) [reply taniumthreatresponse_connector.py]: Instead of directly getting status value now using .get() method.

**Review by alexa-phantom — APPROVED** [review]

**Pull request merged**